src/java/voldemort/server/Host.java

```diff
@@ -0,0 +1,26 @@
+package voldemort.server;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import voldemort.VoldemortApplicationException;
+
+
+public class Host {
+
+    private static InetAddress getLocal() {
+        try {
+            return InetAddress.getLocalHost();
+        } catch(UnknownHostException ex) {
+            throw new VoldemortApplicationException("Error retrieving local host", ex);
+        }
+    }
+
+    public static String getName() {
+        return getLocal().getHostName();
+    }
+
+    public static String getFQDN() {
+        return getLocal().getCanonicalHostName();
+    }
```

Comment:
A reading of the JDK code shows that for an `InetAddress` object of the shape returned by `InetAddress.getLocalHost()`, both `getHostName()` and `getCanonicalHostName()` always return exactly the same thing, except that the latter takes 6 trips through the platform naming system instead of 4.  So you could replace this entire class with a call to `InetAddress.getLocalHost().getHostName()`.